### PR TITLE
Refactor master to control plane

### PR DESCRIPTION
**Pull request by gooney152**
#### What type of PR is this?
/kind clean up
/wg-naming

#### What this PR does / why we need it:
refactor "master" to "controlplane"

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by gooney152** [comment]
references issue #94901

**Comment by gooney152** [comment]
@pohly

**Comment by gooney152** [comment]
@smarterclayton

File: test/e2e/cloud/gcp/ha_master.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ import (
 	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 )
 
-func addMasterReplica(zone string) error {
+func addControlPlaneReplica(zone string) error {
 	framework.Logf(fmt.Sprintf("Adding a new master replica, zone: %s", zone))
```
Comment (BenTheElder) [inline test/e2e/cloud/gcp/ha_master.go:41]: log message still says "master" (There are more of these in the renamed methods)

File: test/e2e/cloud/gcp/ha_master.go [hunk 2]
```diff
@@ -117,23 +117,23 @@ func removeZoneFromZones(zones []string, zone string) []string {
...
 		if err != nil {
 			framework.Logf("Failed to list nodes: %v", err)
 			continue
 		}
 
-		// Filter out nodes that are not master replicas
+		// Filter out nodes that are not controlplane replicas
 		e2enode.Filter(nodes, func(node v1.Node) bool {
-			res, err := regexp.Match(generateMasterRegexp(masterPrefix), ([]byte)(node.Name))
+			res, err := regexp.Match(generateControlPlaneRegexp(masterPrefix), ([]byte)(node.Name))
```
Comment (neolit123) [inline test/e2e/cloud/gcp/ha_master.go:136]: i think we should make broader changes, changing variable names like `masterPrefix` as well in the same PR.
[1]

File: test/e2e/cloud/gcp/ha_master.go [hunk 3]
```diff
@@ -171,7 +171,7 @@ var _ = SIGDescribe("HA-master [Feature:HAMaster]", func() {
 		e2eskipper.SkipUnlessProviderIs("gce")
 		c = f.ClientSet
 		ns = f.Namespace.Name
-		framework.ExpectNoError(waitForMasters(framework.TestContext.CloudConfig.MasterName, c, 1, 10*time.Minute))
+		framework.ExpectNoError(waitForControlPlanes(framework.TestContext.CloudConfig.MasterName, c, 1, 10*time.Minute))
```
Comment (neolit123) [inline test/e2e/cloud/gcp/ha_master.go:174]: [1]
`TestContext.CloudConfig.MasterName` should be safe to change, given the TestContext struct gives no deprecation guarantees to users.

**Pull request closed without merging**